tests/tree_xml_parser_paragraph_processor_tests.py

```diff
@@ -111,15 +111,25 @@ def test_build_hierarchy_markerless(self):
         self.assertEqual(b.label, ['root', 'p1', 'b'])
 
     def test_separate_intro_positive(self):
-        """Positive test case for a separate introductory paragraph"""
+        """Positive test cases for a separate introductory paragraph"""
+        # The typical case:
         nodes = [Node(label=[mtypes.MARKERLESS]), Node(label=['a']),
                  Node(label=['b']), Node(label='1')]
         intro, rest = _ExampleProcessor().separate_intro(nodes)
         self.assertEqual(nodes[0], intro)
         self.assertEqual(nodes[1:], rest)
 
+        # A MARKERLESS node followed by a table node:
+        xml_table = etree.fromstring('<GPOTABLE>stuff</GPOTABLE>')
+        table_node = Node(label=[mtypes.MARKERLESS], source_xml=xml_table)
+        nodes = [Node(label=[mtypes.MARKERLESS, table_node])]
+        intro, rest = _ExampleProcessor().separate_intro(nodes)
+        self.assertEqual(nodes[0], intro)
+        self.assertEqual(nodes[1:], rest)
```

Comment:
Would breaking this into a separate test be easy? Easier to pinpoint failures. Looks like some redundancy could be removed, too